### PR TITLE
Feature/id 123 abandon flow

### PR DESCRIPTION
**Pull request by spinhalf-com**
## Purpose

_Briefly describe the purpose of the change._

Fixes ###-####

## Approach

_Explain how your code addresses the purpose of the change._

## Learning

_Any tips and tricks, blog posts or tools which helped you._

## Checklist

* [x ] I have performed a self-review of my own code
* [x ] I have added relevant logging with appropriate levels to my code
* [x ] I have updated documentation where relevant
* [x ] I have added tests to prove my work
* [x ] I have run an accessibility tool against the changes and applied fixes
* [ ] The team have tested these changes

### CODE REVIEW

File: service-api/module/Application/test/Controller/IdentityControllerTest.php [hunk 1]
```diff
@@ -813,4 +813,52 @@ public static function lpaRemoveData(): array
...
+     */
+    public function testAbandonFlow(
+        string $uuid,
+        array $data,
+        array $response
+    ): void {
+
+        $this->dataImportHandler
+            ->expects($this->once())
+            ->method('updateCaseData');
```
Comment (gregtyler) [inline service-api/module/Application/test/Controller/IdentityControllerTest.php:828]: Worth adding `->with()` to this to make sure it's saving the correct data

File: service-front/module/Application/view/application/pages/abandoned_flow.twig [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    {% include 'layout/id_check_banner.twig' with details_data %}
+    <form method="post" class="govuk-!-width-two-thirds">
+        <div class="govuk-width-container govuk-!-padding-top-5">
+            <h1 class="govuk-heading-xl">
+                {% block title %}
+                    Exit identity check?
+                {% endblock %}
+            </h1>
+            <div class="govuk-form-group">
+                <input type="hidden"  name="last_page" value="{{ last_page }}">
```
Comment (gregtyler) [inline service-front/module/Application/view/application/pages/abandoned_flow.twig:12]: Is `last_page` being used for anything in this form?

File: service-front/module/Application/view/application/pages/abandoned_flow.twig [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                            <label class="govuk-label govuk-radios__label" for="reason-3">
+                                Other
+                            </label>
+                        </div>
+                    </div>
+                </fieldset>
+            </div>
+            <div class="govuk-form-group">
+                <h2 class="govuk-label-wrapper">
+                    <label class="govuk-label govuk-label--m" for="more-detail">
```
Comment (gregtyler) [inline service-front/module/Application/view/application/pages/abandoned_flow.twig:50]:
```suggestion
                    <label class="govuk-label govuk-label--m" for="notes">
```

File: service-front/module/Application/src/Services/SiriusApiService.php [hunk 4]
```diff
@@ -130,4 +130,20 @@ public function searchAddressesByPostcode(string $postcode, Request $request): a
...
+        $response = $this->client->post('/api/v1/identity-check', [
+            'headers' => $this->getAuthHeaders($request),
+            'data' => $data
+        ]);
+
+        return [
+            'status' => $response->getStatusCode(),
+            'error' => json_decode(strval($response->getBody()), true)
+        ];
+    }
```
Comment (gregtyler) [inline service-front/module/Application/src/Services/SiriusApiService.php:148]: This needs a Pact test (in `SiriusApiServicePactTest`) to establish a contract between the services

File: service-front/module/Application/view/application/pages/abandoned_flow.twig [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                    </legend>
+                    {% if form.get('reason').messages %}
+                        <p id="dln-error" class="govuk-error-message">
+                            <span class="govuk-visually-hidden">Error:</span>
+                            {{ form.get('reason').messages | join(', ', ' and ') }}
+                        </p>
+                    {% endif %}
+                    <div class="govuk-radios" data-module="govuk-radios">
+                        <div class="govuk-radios__item">
+                            <input class="govuk-radios__input" id="reason" name="reason" type="radio" value="cd">
```
Comment (gregtyler) [inline service-front/module/Application/view/application/pages/abandoned_flow.twig:28]: We should probably figure out what the business wants to happen with this information, at the moment it's just being ignored

File: service-front/module/Application/view/application/pages/abandoned_flow.twig (hunk 2 above)
Comment (spinhalf-com) [reply service-front/module/Application/view/application/pages/abandoned_flow.twig:12]: It'll be used in the progess tracking feature. This branch was initially going to include the progress tracking, but I decided to split the tickets.

**Review by gregtyler — APPROVED** [review]
Looks good, but a reminder to include the ticket name and semver type in the PR title/body before merging.

**Pull request merged**